SpinCore_pp/ppg/generic.py

```diff
@@ -133,27 +145,19 @@ def generic(
         runBoard()
         run_scans_time_list.append(time.time())
         run_scans_names.append("get data")
-        raw_data = getData(data_length, nPoints, nEchoes, nPhaseSteps)
+        raw_data = getData(int(data_length), nPoints, nEchoes, int(nPhaseSteps))
+        raw_data.astype(float)
         run_scans_time_list.append(time.time())
         run_scans_names.append("shape data")
+        # {{{ create returned data
         data_array = []
         data_array[::] = np.complex128(raw_data[0::2] + 1j * raw_data[1::2])
         dataPoints = float(np.shape(data_array)[0])
         if ret_data is None:
-            if indirect_fields is None:
-                times_dtype = np.double
-            else:
-                # {{{ dtype for structured array
-                times_dtype = np.dtype(
-                    [(indirect_fields[0], np.double), (indirect_fields[1], np.double)]
-                )
-                # }}}
-            mytimes = np.zeros(indirect_len, dtype=times_dtype)
-            time_axis = r_[0:dataPoints] / (SW_kHz * 1e3)
+            time_axis = np.linspace(0.0,nEchoes*int(nPhaseSteps)*acq_time_ms*1e-3,int(dataPoints))
```

Comment:
this change doesn't appear to be justified.

Reply:
added comment - this is consistent with how the other ppgs in the folder save the time axis